src/plugins/data_source_management/public/components/data_source_column/data_source_column.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import { HttpStart, IBasePath, SavedObjectsStart } from 'opensearch-dashboards/public';
+import { EuiBadge, EuiLink } from '@elastic/eui';
+import React from 'react';
+import {
+  IndexPatternTableColumn,
+  IndexPatternTableRecord,
+} from '../../../../index_pattern_management/public';
+import { getDataSources } from '../utils';
+import { DataSourceTableItem } from '../../types';
+
+type DataSourceMap = Map<string, DataSourceTableItem>;
+
+export class DataSourceColumn implements IndexPatternTableColumn<DataSourceMap> {
+  public readonly id: string = 'data_source';
+  public data?: DataSourceMap;
+  private basePath?: IBasePath;
+
+  public euiColumn = {
+    field: 'referenceId',
+    name: i18n.translate('dataSource.management.dataSourceColumn', {
+      defaultMessage: 'Data Source',
+    }),
+    render: (referenceId: string, index: IndexPatternTableRecord) => {
+      if (!referenceId) {
+        return null;
+      }
+
+      const dataSource = this.data?.get(referenceId);
+      if (!dataSource) {
+        // Index pattern has the referenceId but data source not found.
+        return <EuiBadge color="danger">Deleted</EuiBadge>;
+      }
+
+      const { id, title } = dataSource;
+      const relativePath = this.basePath?.prepend(
+        `/app/management/opensearch-dashboards/dataSources/${encodeURIComponent(id)}`
+      );
+      return <EuiLink href={relativePath}>{title}</EuiLink>;
```

Comment:
So if `basePath` is undefined, the href would be to `./undefined`?

Reply:
thanks for noticing, forgot to handle that